core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java

```diff
@@ -168,23 +168,31 @@ public void initialize(String name, Map<String, String> unresolved) {
     Map<String, String> baseHeaders = configHeaders(mergedProps);
 
     this.sessions = newSessionCache(mergedProps);
-    this.refreshAuthByDefault =
+    this.keepTokenRefreshed =
         PropertyUtil.propertyAsBoolean(
             mergedProps,
-            CatalogProperties.AUTH_DEFAULT_REFRESH_ENABLED,
-            CatalogProperties.AUTH_DEFAULT_REFRESH_ENABLED_DEFAULT);
+            OAuth2Properties.TOKEN_REFRESH_ENABLED,
+            OAuth2Properties.TOKEN_REFRESH_ENABLED_DEFAULT);
     this.client = clientBuilder.apply(mergedProps);
     this.paths = ResourcePaths.forCatalogProperties(mergedProps);
 
     this.catalogAuth = new AuthSession(baseHeaders, null, null, credential, scope);
     if (authResponse != null) {
       this.catalogAuth =
           AuthSession.fromTokenResponse(
-              client, tokenRefreshExecutor(), authResponse, startTimeMillis, catalogAuth);
+              client,
+              keepTokenRefreshed ? tokenRefreshExecutor() : null,
```

Comment:
Does the other branch of a ternary operator get called? If so, this would start the thread even if it isn't used.

Seems like the behavior would be more obvious if we embedded this in `tokenRefreshExecutor()` so that if `keepTokenRefreshed` is false, it never starts a thread and returns `null` itself.

Reply:
I've pulled the check into `tokenRefreshExecutor()` so that we don't have to unnecessarily init a `ScheduledExecutorService`.